zoo/board_games/mcts_bot.py

```diff
@@ -177,52 +237,73 @@ def best_action(self, simulations_number=None, total_simulation_seconds=None):
             Returns the best children node, and can get action from Node.best_action_index.
         """
 
+        # The search cost is determined by either the maximum number of simulations or the longest simulation time.
         if simulations_number is None:
             assert (total_simulation_seconds is not None)
             end_time = time.time() + total_simulation_seconds
             while True:
+                # get the leaf node.
                 v = self._tree_policy()
+                # rollout from the leaf node.
                 reward = v.rollout()
+                # backpropagate from the leaf node to the root node.
                 v.backpropagate(reward)
                 if time.time() > end_time:
                     break
         else:
             for i in range(0, simulations_number):
                 # print('****simlulation-{}****'.format(i))
+                # get the leaf node.
                 v = self._tree_policy()
+                # rollout from the leaf node.
                 reward = v.rollout()
                 # print('reward={}\n'.format(reward))
+                # backpropagate from the leaf node to the root node.
                 v.backpropagate(reward)
         # to select best child go for exploitation only
+        # 这里和算法理论里选择最高访问次数似乎不太一样？？？？？？？？？？？？？？？？？？
         return self.root.best_child(c_param=0.)
 
+    #
     def _tree_policy(self):
         """
         Overview:
-            The policy used to select action to rollout a episode.
+            This function implement the tree search from the root node to the leaf node, which is visited for the first time or is the terminal node.
         """
         current_node = self.root
         while not current_node.is_terminal_node():
             if not current_node.is_fully_expanded():
+                # choose a child node which has not been visited before
                 return current_node.expand()
             else:
                 current_node = current_node.best_child()
         return current_node
 
 
 class MCTSBot:
-
+    """
```

Comment:
"""
         Initialize the Monte Carlo Tree Search Bot.

         Arguments:
             - ENV: The environment in which the MCTS bot operates.
             - cfg: Configuration parameters for the bot.
             - bot_name (str): The name of the bot.
             - num_simulation (int): Number of simulations to perform for each move. Default is 10000.
         """